webpack/components/Content/ContentDetails.js

```diff
@@ -0,0 +1,104 @@
+//import BreadcrumbsBar from 'foremanReact/components/BreadcrumbBar';
+import { PropTypes } from 'prop-types';
+import { LoadingState } from '../../move_to_pf/LoadingState';
+//import api from '../../../services/api';
+import ModuleStreamDetailInfo from '../../scenes/ModuleStreams/Details/ModuleStreamDetailInfo';
+import ModuleStreamDetailRepositories from '../../scenes/ModuleStreams/Details/Repositories/ModuleStreamDetailRepositories';
+import ModuleStreamDetailArtifacts from '../../scenes/ModuleStreams/Details/ModuleStreamDetailArtifacts';
+import ModuleStreamDetailProfiles from '../../scenes/ModuleStreams/Details/Profiles/ModuleStreamDetailProfiles';
+
+const ContentDetails = props => {
+  const { contentDetails, tabLabels } = props;
+  const {
+    loading, profiles, repositories, artifacts,
+  } = contentDetails;
+
+  const tabHeaders = tabLabelsProp => {
+    const tabs = tabLabelsProp.map((label, index) =>
+      <NavItem eventKey={index+3}>
+        <div>{label}</div>
+      </NavItem>
+    );
+    return (
+      tabs
+    );
+  };
+  
+  return (
+    <div>
+      <LoadingState
+        loading={loading}
+        loadingText={__('Loading')}
+      >
+        <TabContainer id="content-tabs-container" defaultActiveKey={1}>
+          <Grid bsClass="container-fluid">
+            <Row>
+              <Col sm={12}>
+                <Nav bsClass="nav nav-tabs">
+                  <NavItem eventKey={1}>
+                    <div>{__('Details')}</div>
+                  </NavItem>
+                  <NavItem eventKey={2}>
+                    <div>{__('Repositories')}</div>
+                  </NavItem>
+                  {tabHeaders(tabLabels)}
+                </Nav>
+              </Col>
+            </Row>
+            <TabContent animation={false}>
+              <TabPane eventKey={1}>
+                <Row>
+                  <Col sm={12}>
+                    <ModuleStreamDetailInfo moduleStreamDetails={contentDetails} />
+                  </Col>
+                </Row>
+              </TabPane>
+              <TabPane eventKey={2}>
+                <Row>
+                  <Col sm={12}>
+                    {repositories && repositories.length ?
+                      <ModuleStreamDetailRepositories repositories={repositories} /> :
+                      __('No repositories to show')}
+                  </Col>
+                </Row>
+              </TabPane>
+              <TabPane eventKey={3}>
+                <Row>
+                  <Col sm={12}>
+                    {profiles && profiles.length ?
+                      <ModuleStreamDetailProfiles profiles={profiles} /> :
+                      __('No profiles to show')}
+                  </Col>
+                </Row>
+              </TabPane>
+              <TabPane eventKey={4}>
+                <Row>
+                  <Col sm={12}>
+                    {artifacts && artifacts.length ?
+                      <ModuleStreamDetailArtifacts artifacts={artifacts} /> :
```

Comment:
Here, for ModuleStreamDetailProfiles and ModuleStreamDetailArtifacts, even if we generate it here, we would need to import these pages in this common component. How do we go about achieving dynamic imports based on what we are trying to render? I didn't find any good resources on that and I basically verified that React's `SyntaxError: 'import' and 'export' may only appear at the top level` works.. :smile: